data/functions.sql

```diff
@@ -1264,3 +1264,157 @@ BEGIN
+        END IF;
+
+    ELSE
+            SELECT hstore(ARRAY[
+                'ne_name_ar', sp.name_ar,
+                'ne_name_bn', sp.name_bn,
+                'ne_name_de', sp.name_de,
+                'ne_name_en', sp.name_en,
+                'ne_name_es', sp.name_es,
+                'ne_name_fa', sp.name_fa,
+                'ne_name_fr', sp.name_fr,
+                'ne_name_el', sp.name_el,
+                'ne_name_he', sp.name_he,
+                'ne_name_hi', sp.name_hi,
+                'ne_name_hu', sp.name_hu,
+                'ne_name_id', sp.name_id,
+                'ne_name_it', sp.name_it,
+                'ne_name_ja', sp.name_ja,
+                'ne_name_ko', sp.name_ko,
+                'ne_name_nl', sp.name_nl,
+                'ne_name_pl', sp.name_pl,
+                'ne_name_pt', sp.name_pt,
+                'ne_name_ru', sp.name_ru,
+                'ne_name_sv', sp.name_sv,
+                'ne_name_tr', sp.name_tr,
+                'ne_name_uk', sp.name_uk,
+                'ne_name_ur', sp.name_ur,
+                'ne_name_vi', sp.name_vi,
+                'ne_name_zh', sp.name_zh,
+                'ne_name_zht', sp.name_zht
+                ]) INTO ne_name_tags
+            FROM ne_10m_admin_1_states_provinces sp
+            WHERE sp.wikidataid = wikidata_id;
+
+        -- finally, try localities
+        IF NOT FOUND THEN
+            SELECT hstore(ARRAY[
+                'ne_name_ar', pp.name_ar,
+                'ne_name_bn', pp.name_bn,
+                'ne_name_de', pp.name_de,
+                'ne_name_en', pp.name_en,
+                'ne_name_es', pp.name_es,
+                'ne_name_fa', pp.name_fa,
+                'ne_name_fr', pp.name_fr,
+                'ne_name_el', pp.name_el,
+                'ne_name_he', pp.name_he,
+                'ne_name_hi', pp.name_hi,
+                'ne_name_hu', pp.name_hu,
+                'ne_name_id', pp.name_id,
+                'ne_name_it', pp.name_it,
+                'ne_name_ja', pp.name_ja,
+                'ne_name_ko', pp.name_ko,
+                'ne_name_nl', pp.name_nl,
+                'ne_name_pl', pp.name_pl,
+                'ne_name_pt', pp.name_pt,
+                'ne_name_ru', pp.name_ru,
+                'ne_name_sv', pp.name_sv,
+                'ne_name_tr', pp.name_tr,
+                'ne_name_uk', pp.name_uk,
+                'ne_name_ur', pp.name_ur,
+                'ne_name_vi', pp.name_vi,
+                'ne_name_zh', pp.name_zh,
+                'ne_name_zht', pp.name_zht
+                ]) INTO ne_name_tags
+            FROM ne_10m_populated_places pp
+            WHERE pp.wikidataid = wikidata_id;
+        END IF;
+    END IF;
+
+      -- return an empty JSONB rather than null, so that it can be safely
+      -- concatenated with whatever other JSONB rather than needing a check for
+      -- null.
+    IF NOT FOUND THEN
+        RETURN '{}'::jsonb;
+    END IF;
+    RETURN to_jsonb(coalesce(ne_name_tags, ''::hstore));
```

Comment:
oh does this coalesce turn an array into a map?   if so, then ignore my previous comment about arrays